src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs

```diff
@@ -36,46 +36,102 @@
         {
             var isAsync = instrumentedMethodCall.IsAsync;
 
-            var inputObject = instrumentedMethodCall.MethodCall.MethodArguments[0];
-            var lambdaContext = instrumentedMethodCall.MethodCall.MethodArguments[1]; // TODO handle case where this doesn't exist
+            object lambdaContext = null;
+            object inputObject = null;
+            AwsLambdaEventType eventType = AwsLambdaEventType.Unknown;
 
-            var functionNameGetter = _getFunctionNameFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "FunctionName");
-            var functionVersionGetter = _getFunctionVersionFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "FunctionVersion");
-            var requestIdGetter = _getAwsRequestIdFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "AwsRequestId");
-            var functionArnGetter = _getInvokedFunctionArnFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "InvokedFunctionArn");
+            foreach (var arg in instrumentedMethodCall.MethodCall.MethodArguments)
+            {
+                // TODO: I don't like this approach, but our options are limited without access to the interface
+                if (arg.GetType().FullName.Contains("LambdaContext"))
+                {
+                    if (lambdaContext == null)
+                    {
+                        lambdaContext = arg;
+                    }
+                    else
+                    {
+                        agent.Logger.Log(Agent.Extensions.Logging.Level.Warn, $"Found multiple Lambda contexts, will use the first found");
+                    }
+                }
+                else
+                {
+                    var fullName = arg.GetType().FullName;
+                    agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"input object fullname = {fullName}");
+                    eventType = fullName.ToEventType();
+                    if (eventType != AwsLambdaEventType.Unknown)
+                    {
+                        inputObject = arg;
+                        agent.Logger.Log(Agent.Extensions.Logging.Level.Debug, $"eventType = {eventType}");
```

Comment:
This seems like a false positive, but I did not look deep enough to fully confirm it.

Reply:
This came up in a prior PR and I'm still puzzled by it. It's just an enum value and no different than the others in that enum. I'm not sure what it thinks is "private" data. Maybe because it has "email" in the name?